packages/shared/src/hooks/useCompanionTrigger.tsx

```diff
@@ -71,33 +71,53 @@ export default function useCompanionTrigger(
     closeLazyModal();
   }, [closeLazyModal]);
 
-  const activateCompanion = useCallback(async () => {
-    await requestContentScripts();
-    closeModal();
-  }, [requestContentScripts, closeModal]);
-
   const openArticle: OpenArticle = useCallback(
-    (data) => async () => {
+    (data: CompanionTriggerProps) => async (redirectUrl?: string) => {
       await customPostClickHandler(
         data.post,
         data.index,
         data.row,
         data.column,
       );
 
+      if (redirectUrl) {
+        window.open(redirectUrl, '_blank');
+      }
+
       closeModal();
     },
     [customPostClickHandler, closeModal],
   );
 
+  const activateCompanion = useCallback(
+    (data) => async (redirectUrl?: string) => {
+      await requestContentScripts(true);
+
+      await customPostClickHandler(
+        data.post,
+        data.index,
+        data.row,
+        data.column,
+      );
+
+      if (redirectUrl) {
+        window.open(redirectUrl, '_blank');
+      }
+
+      closeModal();
+    },
+    [requestContentScripts, customPostClickHandler, closeModal],
+  );
+
   const openModal = useCallback(
     (data?: CompanionTriggerProps) => {
       openLazyModal({
         type: LazyModal.CompanionModal,
         props: {
           url: data?.post?.permalink,
           onReadArticleClick: openArticle(data),
-          onActivateCompanion: activateCompanion,
+          onActivateCompanion: activateCompanion(data),
+          onRequestClose: closeModal,
```

Comment:
I am a bit at a loss regarding why this line causes a TS error -- `onRequestClose` should be an accepted attribute here based on how `LazyModalProps` is defined: https://github.com/dailydotdev/apps/blob/ecdf285060500aef2ea8b981ca12d48f6f03e975/packages/shared/src/components/modals/common/Modal.tsx#L32-L35

Reply:
You shouldn't need to manually push that.
the idea behind the lazymodal is that it can trigger that itself

Reply:
Things get a bit tricky when you need a custom close button like the situation we have here. What are your thoughts on handling this?

Reply:
You don't really need a custom one though, the action already exists.
You can just use it on the specific modal right

Reply:
you're right, of course, I think I had a brain fart here